app/Http/Controllers/Service/Admin/TradersController.php

```diff
@@ -152,11 +153,27 @@ public function update(AdminNewUpdateTraderRequest $request, int $id)
                 // find the trader
                 $t = Trader::findOrFail($id);
 
-                // Update it
-                $t->fill([
+                // make an array of things to fill
+                $fillArray = [
                     'market_id' => $request->input('market'),
                     'name' => $request->input('name'),
-                ])->save();
+                ];
+
+                // If form's disabled input and Trader's disabled_at are in different states
+                if (boolval($request->input('disabled')) xor
```

Comment:
This seems a little convoluted but if it is the effect you are after, the comments seem clear enough.

Reply:
I built a huge truth table for flipping things based on two condistions and xor seemed to be the best match.